datadog_checks_base/tests/test_db_statements.py

```diff
@@ -109,6 +133,80 @@ def key(row):
         assert 1 == len(sm.compute_derivative_rows(rows3, metrics, key=key))  # only 1 row computed
         assert 2 == len(sm.compute_derivative_rows(rows4, metrics, key=key))  # both rows computed
 
+    def test_compute_derivative_rows_with_duplicates(self, datadog_agent):
```

Comment:
That's one new test but depending on how the discussions on the other topics go, I'd probably want to add a direct test for `merge_duplicate_rows` (which is only tested indirectly here right now).